src/Explorer/Tree/Collection.ts

```diff
@@ -1331,6 +1331,20 @@ export default class Collection implements ViewModels.Collection {
     return this.container.findDatabaseWithId(this.databaseId);
   }
 
+  public async loadAutopilotOfferWithRetry(): Promise<DataModels.Offer> {
+    let retryCount = 0;
```

Comment:
We use p-retry in other places which should do all this for you https://www.npmjs.com/package/p-retry

Reply:
I checked it out. It gets a little cluttered when 1) Specific retry conditions need to be checked 2) We still have to add a sleep/delay/settimeout for delays between retries. i don't see it being much better than the existing implementation

Reply:
synced offline, going with the existing pattern